File: tests/Python/AffineTest.py
from Affine import *
import unittest

class TestAffine(unittest.TestCase):
    
    def test_sanity(self):
        c = Affine(3, 4)
        self.assertTrue(c.encrypt("hello")=="zqllu")
        self.assertTrue(c.decrypt("zqllu")=="hello")


if __name__ == "__main__":
    unittest.main()

File: tests/Ruby/AffineTest.rb
require '../../lib/Ruby/Affine.rb'
require 'test/unit'

class AffineTest < Test::Unit::TestCase

    def test_sanity
        c = Affine.new(3, 4)
        assert_equal("zqllu", c.encrypt("hello"))
        assert_equal("hello", c.decrypt("zqllu"))
    end
end
